org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/AbstractImageBuilder.java

```diff
@@ -190,6 +190,15 @@ public void acceptResult(CompilationResult result) {
 				if (result.checkSecondaryTypes && !qualifiedTypeName.equals(compilationUnit.initialTypeName))
 					acceptSecondaryType(classFile);
 			}
+			for (int j = 0, l = this.javaBuilder.participants == null ? 0 : this.javaBuilder.participants.length; j < l; j++) {
+				BuildContext buildContext = new CompilationParticipantResult(compilationUnit,
+						this.compilationGroup == CompilationGroup.TEST);
+				Optional<byte[]> postProcessingResult = this.javaBuilder.participants[j].postProcess(buildContext,
+						new ByteArrayInputStream(classFile.getBytes()));
+				if (postProcessingResult.isPresent()) {
```

Comment:
Please add `recordParticipantResult(buildContext);` call, to record dependencies / problems (also if the class file was NOT modified.